### ISSUE TITLE
Clang accepts invalid program involving multiple template parameter packs

### ISSUE THREAD

**Issue by ranaanoop** [llvm/llvm-project#58668]
The following invalid program is accepted by clang and gcc but correctly rejected by msvc: [Demo](https://godbolt.org/z/7WT6fe5n4).

```
template<typename... T1, typename... T2, typename... T3>
void foo(T1&&...args1, T2&&... args2, T3&&... args3) {
    std::cout << "args1:\n", ((std::cout << args1 << " "), ...);
    std::cout << "args2:\n", ((std::cout << args2 << " "), ...);
    std::cout << "args3:\n", ((std::cout << args3 << " "), ...);
}

int main(int argc, char** argv)
{
    foo(1,2,3,4,5,6); //gcc and clang compiles this while msvc correctly rejects this
};
```

**Comment by shafik** [comment]
MSVC looks correct here, I believe this is [temp.param p14](https://eel.is/c++draft/temp.param#14):

> ... A template parameter pack of a function template shall not be followed by another template parameter unless that template parameter can be deduced from the parameter-type-list ([[dcl.fct]](https://eel.is/c++draft/dcl.fct)) of the function template or has a default argument ([[temp.deduct]](https://eel.is/c++draft/temp.deduct))[.](https://eel.is/c++draft/temp.param#14.sentence-3) ...

**Comment by ranaanoop** [comment]
@shafik Yes, but if we remove the third parameter `T3` then msvc also starts compiling the program even though it is still ill-formed. [Demo](https://godbolt.org/z/qacMzoT3q).

**Comment by Halalaluyafail3** [comment]
> [@shafik](https://github.com/shafik) Yes, but if we remove the third parameter `T3` then msvc also starts compiling the program even though it is still ill-formed. [Demo](https://godbolt.org/z/qacMzoT3q).

`foo<int>(1,2)` in that example can be used to give arguments to both packs.
